test/integration/compAnchorIntegration.js

```diff
@@ -0,0 +1,559 @@
+      relayer: operator,
+      fee,
+      refund,
+      chainID: originDeposit.chainID,
+      roots: [destNativeRoot, ...destNeighborRoots],
+      // private
+      nullifier: originDeposit.nullifier,
+      secret: originDeposit.secret,
+      pathElements: path_elements,
+      pathIndices: path_index,
+      diffs: [destNativeRoot, ...destNeighborRoots].map(r => {
+        return F.sub(
+          Scalar.fromString(`${r}`),
+          Scalar.fromString(`${destNeighborRoots[0]}`),
+        ).toString();
+      }),
+    };
+
+    let wtns = await createWitness(input);
+
+    let res = await snarkjs.groth16.prove('test/fixtures/circuit_final.zkey', wtns);
+    proof = res.proof;
+    publicSignals = res.publicSignals;
+    let vKey = await snarkjs.zKey.exportVerificationKey('test/fixtures/circuit_final.zkey');
+    res = await snarkjs.groth16.verify(vKey, publicSignals, proof);
+    assert.strictEqual(res, true);
+
+    let args = [
+      helpers.createRootsBytes(input.roots),
+      helpers.toFixedHex(input.nullifierHash),
+      helpers.toFixedHex(input.recipient, 20),
+      helpers.toFixedHex(input.relayer, 20),
+      helpers.toFixedHex(input.fee),
+      helpers.toFixedHex(input.refund),
+    ];
+
+    let proofEncoded = await helpers.generateWithdrawProofCallData(proof, publicSignals);
+
+    /*
+    *  user1 withdraw on dest chain
+    */
+    ({ logs } = await DestChainLinkableAnchorInstance.withdraw
+      (`0x${proofEncoded}`, ...args, { from: input.relayer, gasPrice: '0' }));
+
+    let balanceDestAnchorAfter = await destWrapperToken.balanceOf(DestChainLinkableAnchorInstance.address);
+    let balanceOperatorAfter = await destWrapperToken.balanceOf(input.relayer);
+    let balanceUser1AfterUnwrap = await destWrapperToken.balanceOf(user1);
+    const feeBN = toBN(fee.toString())
+    assert.strictEqual(balanceDestAnchorAfter.toString(), toBN(0).toString());
+    assert.strictEqual(balanceOperatorAfter.toString(), balanceOperatorBefore.add(feeBN).toString());
+    assert.strictEqual(balanceUser1AfterUnwrap.toString(), balanceReceiverBefore.add(toBN(tokenDenomination)).sub(feeBN).toString());
+    assert.strictEqual((await destWrapperToken.totalSupply()).toString(), tokenDenomination.toString());
+    isSpent = await DestChainLinkableAnchorInstance.isSpent(helpers.toFixedHex(input.nullifierHash));
+    assert(isSpent);
+
+    //originToken is voted to be approved for token wrapper on origin chain
+    await helpers.addTokenToWrapper(destGov, destWrapperToken, destToken, bravoAdmin, states);
+    /*
+    *  user2 wraps destToken for destWrapperToken
+    */
+    await destToken.mint(user2, initialTokenMintAmount);
+    // approve tokenwrapper to transfer destTokens from user2
+    await destToken.approve(destWrapperToken.address, initialTokenMintAmount, {from: user2});
+    // increase allowance for user1 to burn
+    await destWrapperToken.approve(DestChainLinkableAnchorInstance.address, initialTokenMintAmount, {from: user1});
```

Comment:
Can you explain this one more? Increase allowance for a burn? If this is removed does the test fail?

Reply:
Yes the test fails if this is removed. Burning requires an allowance similar to transferFrom. I am working on ways to optimize all these approvals for deposit-and-wrap and something similar with withdrawing unwrapping and burning.